### PR TITLE
Added: VGG-16 configurations and model

### PR DESCRIPTION
**Pull request by miguelCalado**
# Description

Added another model, VGG-16, and necessary configurations to train on the ImageNet dataset to the respective Image classification folder.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] A new research paper code implementation

## Tests and configurations

I trained and tested the model myself on the ImageNet dataset with the following configurations, which are similar to the original paper implementation:

| Parameter                | value  |
| ------------------------ | ------ |
| Batch size           | `256`   |
| Epochs               | `90`   |
| Learning Rate                   | `0.01`  |
| Momentum             | `0.9`  |
| Learning Rate Schedule | `Stepwise` |
| Boundaries | `[30, 60]` |
| Learning rate multipler | `0.1` | 
| Weight decay | `1e-4` |
| Dropout | `0.5` |
| Label Smoothing | `0` |

They can also be found extensively in the `examples/vgg16/gpu.yaml` file. 

After 90 epochs, I obtained a validation Top-1 error of 72.8% and a Top-5 error of 91.1%. which confirms previous results [1, 2]. You can also check the performance of the network throughout the training by accessing the [Weights and Biases report](https://wandb.ai/gohan/train_imagenet/runs/1zokrwti).

On the ImageNet test set, I got 71.1% and 90.3% of Top-1 and Top-5 errors, which again confirms previous results. I made available the weights for the fully trained model in [Google Drive](https://drive.google.com/drive/folders/1rAPS1wPz2nHT1rw_mZ_QqbjMSFxdrl63?usp=sharing). 

To train the network you just need to run:

`python3 classifier_trainer.py --mode=train_and_eval --model_type=vgg --dataset=imagenet --model_dir=vgg16 --data_dir=imagenet-data/ILSVRC/tfrecords/ --config_file=configs/examplesvgg16/gpu.yaml`

Refences:

- [1] - https://github.com/keras-team/keras-applications
- [2] - https://pytorch.org/hub/pytorch_vision_vgg/

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by saberkun** [comment]
Very sorry. We are in the process to move the image_classification/ project to the legacy/ folder.
Actually you can see a copy in the legacy/ folder. Overall, the change looks good but could you change the PR to update files in the legacy/image_classification?

**Comment by miguelCalado** [comment]
No problem @saberkun, it actually gave me more time to add more documentation!

File: official/legacy/image_classification/configs/configs.py [hunk 1]
```diff
@@ -92,12 +93,44 @@ class ResNetImagenetConfig(base_configs.ExperimentConfig):
   model: base_configs.ModelConfig = resnet_config.ResNetModelConfig()
 
 
+@dataclasses.dataclass
```
Comment (saberkun) [inline official/legacy/image_classification/configs/configs.py:96]: Here are the pylints reported by CI:

FAIL: Found 24 non-whitelited pylint errors:
official/legacy/image_classification/vgg16/vgg_config.py:30: [E0001(syntax-error), ] invalid syntax

official/legacy/image_classification/configs/configs.py:42: [E0001(syntax-error), ] invalid syntax

official/legacy/image_classification/vgg16/vgg_model.py:33: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:43: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:56: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:66: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:79: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:89: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:99: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:112: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:122: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:132: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:145: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:155: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:165: [C0301(line-too-long), ] Line too long (81/80)

official/legacy/image_classification/vgg16/vgg_model.py:177: [C0301(line-too-long), ] Line too long (84/80)

official/legacy/image_classification/vgg16/vgg_model.py:184: [C0301(line-too-long), ] Line too long (84/80)

official/legacy/image_classification/vgg16/vgg_model.py:191: [C0301(line-too-long), ] Line too long (84/80)

official/legacy/image_classification/classifier_trainer_test.py:28: [C0301(line-too-long), ] Line too long (84/80)

official/legacy/image_classification/classifier_trainer_test.py:69: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer_test.py:102: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer_test.py:103: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer.py:134: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer.py:425: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

Could you fix them? 
Note that, pylint can be wrong so if you feel any does not make sense, please ignore them

**Review by saberkun — CHANGES_REQUESTED** [review]

File: official/legacy/image_classification/configs/configs.py (hunk 1 above)
Comment (miguelCalado) [reply official/legacy/image_classification/configs/configs.py:96]: Yes, just fixed them. The following I couldn't find a solution, so I just ignored them like you said:

official/legacy/image_classification/classifier_trainer_test.py:102: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer_test.py:103: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer.py:134: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

official/legacy/image_classification/classifier_trainer.py:425: [C0326(bad-whitespace), ] No space allowed around keyword argument assignment

**Review by saberkun — APPROVED** [review]
Please wait for merging through an internal system.

**Comment by saberkun** [comment]
Thanks. The code is merged.
Just note that, we have consolidated all vision tasks to the official/vision/beta folder now and which will soon merge the content to the vision/ folder.

**Comment by miguelCalado** [comment]
Awesome, thanks for pointing that out. Looking forward to contributing some more!
